File: objectivec/GithubappObjC/DetailsInteractor.m
//
//  DetailViewInteractor.m
//  GithubappObjC
//
//  Created by user124076 on 11.01.17.
//  Copyright © 2017 user124076. All rights reserved.
//
#import "DetailsInteractor.h"

@implementation DetailsInteractor

-(void) requestModel{
    [[self presenter] sendModel:self.model];
}

-(void)configureModel:(NSDictionary *)model{
    self.model = model;
}

@end

File: swift/GithubApp/DetailsInteractor.swift
//
//  RepoDetailsInteractor.swift
//  GithubApp
//
//  Created by user124076 on 08.01.17.
//  Copyright © 2017 user124076. All rights reserved.
//

import Foundation

protocol DetailsInteractorOutput: class {
    func sendModel(_ model: Model)
}

protocol DetailsInteractorInput: class {
    func requestModel()
    func configureModel(_ model: Model)
}

class DetailsInteractor : DetailsInteractorInput {
   
    weak var presenter: DetailsInteractorOutput!
    
    var model: Model?
    
    func configureModel(_ model: Model) {
        self.model = model
    }
    
    func requestModel() {
        self.presenter.sendModel(model!)
        
    }
}
